pkg/build/builder/dockerutil.go

```diff
@@ -126,6 +126,27 @@ func readInt64(filePath string) (int64, error) {
 	return val, nil
 }
 
+func readMaxStringOrInt64(filePath string) (int64, error) {
+	val, err := readInt64(filePath)
+	if err == nil {
+		return val, nil
+	}
+	if os.IsNotExist(err) {
+		return val, err
+	}
+	// see if we have a string called 'max' for the value
+	data, err2 := ioutil.ReadFile(filePath)
```

Comment:
So basically you're reading the file twice -- this is redundant.

I suggest to separate out the portion of readInt64 that parses the value and re-use it from here.

Something like this maybe

```diff
diff --git a/pkg/build/builder/dockerutil.go b/pkg/build/builder/dockerutil.go
index 764e06655..1ef30681a 100644
--- a/pkg/build/builder/dockerutil.go
+++ b/pkg/build/builder/dockerutil.go
@@ -111,7 +111,24 @@ func readInt64(filePath string) (int64, error) {
        if err != nil {
                return -1, err
        }
+
+       return parseInt64(strings.TrimSpace(string(data)))
+}
+
+func readMaxStringOrInt64(filePath string) (int64, error) {
+       data, err := ioutil.ReadFile(filePath)
+       if err != nil {
+               return -1, err
+       }
        s := strings.TrimSpace(string(data))
+       if s == "max" {
+               return math.MaxInt64, nil
+       }
+
+       return parseInt64(s)
+}
+
+func parseInt64(s string) (int64, error) {
        val, err := strconv.ParseInt(s, 10, 64)
        // overflow errors are ok, we'll get return a math.MaxInt64 value which is more
        // than enough anyway.  For underflow we'll return MinInt64 and the error.
```